doc/pachctl/pachctl_update-pipeline.md

```diff
@@ -0,0 +1,243 @@
+## ./pachctl update-pipeline
+
+Update an existing Pachyderm pipeline.
```

Comment:
Does it make sense to have the full pipeline spec here? That adds yet another place where we have to keep docs in sync every time the pipeline spec is changed. I'm not sure what the best answer is here, thought @jdoliner